docs/src/main/asciidoc/kafka.adoc

```diff
@@ -3,438 +3,980 @@ This guide is maintained in the main Quarkus repository
+    @Outgoing("price-out")
+    public double process(double price) {
+        return price * CONVERSION_RATE;
+    }
 
-[source, yaml]
+}
 ----
-version: '2'
 
-services:
+The parameter of the `process` method is the incoming message payload, whereas the return will be used as the outgoing messaage payload. 
+Previously mentioned signatures for parameter and return types, as described in previous sections are supported, such as `Message<T>`, `Record<K, V>` etc.
 
-  zookeeper:
-    image: quay.io/strimzi/kafka:0.23.0-kafka-2.8.0
-    command: [
-      "sh", "-c",
-      "bin/zookeeper-server-start.sh config/zookeeper.properties"
-    ]
-    ports:
-      - "2181:2181"
-    environment:
-      LOG_DIR: /tmp/logs
+You can apply asynchronous stream processing by consuming and returning reactive stream `Multi<T>` type:
 
-  kafka:
-    image: quay.io/strimzi/kafka:0.23.0-kafka-2.8.0
-    command: [
-      "sh", "-c",
-      "bin/kafka-server-start.sh config/server.properties --override listeners=$${KAFKA_LISTENERS} --override advertised.listeners=$${KAFKA_ADVERTISED_LISTENERS} --override zookeeper.connect=$${KAFKA_ZOOKEEPER_CONNECT}"
-    ]
-    depends_on:
-      - zookeeper
-    ports:
-      - "9092:9092"
-    environment:
-      LOG_DIR: "/tmp/logs"
-      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://localhost:9092
-      KAFKA_LISTENERS: PLAINTEXT://0.0.0.0:9092
-      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+[source,java]
 ----
+import javax.enterprise.context.ApplicationScoped;
 
-Once created, run `docker-compose up`.
-
-NOTE: This is a development cluster, do not use in production.
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
 
-You can build and run the application in JVM mode with:
+import io.smallrye.mutiny.Multi;
 
-[source, bash]
-----
-./mvnw package
-java -jar target/quarkus-app/quarkus-run.jar
-----
+@ApplicationScoped
+public class PriceProcessor {
 
-NOTE: By default, the application tries to connect to a Kafka broker listening at `localhost:9092`.
-You can configure the bootstrap server using: `java -Dkafka.bootstrap.servers=... -jar target/quarkus-app/quarkus-run.jar`
+    private static final double CONVERSION_RATE = 0.88;
 
-You can build and run the native executable with:
+    @Incoming("price-in")
+    @Outgoing("price-out")
+    public Multi<Double> process(Multi<Integer> prices) {
+        return prices.filter(p -> p > 100).map(p -> p * CONVERSION_RATE);
+    }
 
-[source,bash]
-----
-./mvnw package -Pnative
-./target/kafka-quickstart-1.0.0-SNAPSHOT-runner -Dkafka.bootstrap.servers=localhost:9092
+}
 ----
 
-== Imperative usage
-
-Sometimes, you need to have an imperative way of sending messages.
-
-For example, if you need to send a message to a stream, from inside a REST endpoint, when receiving a POST request.
-In this case, you cannot use `@Outgoing` because your method has parameters.
+== Accessing Kafka clients directly
```

Comment:
Add an anchor as this should be referenced from the getting started guide.

Reply:
Added the reference to the getting started guide